libs/mimir/src/adapters/secondary/elasticsearch/internal.rs

```diff
@@ -828,27 +828,44 @@ impl ElasticsearchStorage {
     pub(super) async fn force_merge(
         &self,
         indices: &[&str],
-        max_num_segments: i64,
+        max_num_segments: Option<i64>,
+        timeout: Duration,
+        allow_timeout: bool,
     ) -> Result<(), Error> {
-        let response = self
-            .client
-            .indices()
+        let indices_client = self.client.indices();
+
+        let request = indices_client
             .forcemerge(IndicesForcemergeParts::Index(indices))
-            .max_num_segments(max_num_segments)
-            // .request_timeout(self.config.timeout) This call is not using timeout because
-            // it can take a long time and would require the timeout to become very large,
-            // and meaningless for other operations.
-            .send()
-            .await
+            .request_timeout(timeout);
+
+        let request = {
+            if let Some(max_num_segments) = max_num_segments {
+                request.max_num_segments(max_num_segments)
+            } else {
+                request
+            }
+        };
+
+        let response = request.send().await;
+
+        // The forcemerge operation can be very long if a large number of segments have to be
+        // merged, in such a case the user may set `allow_timeout` to true in order to let the
+        // operation run in background.
+        if allow_timeout && matches!(&response, Err(err) if err.is_timeout()) {
+            warn!(
+                "forcemerge query timeout after {:?} on indices {}, it will continue running in background",
+                timeout,
+                indices.iter().copied().collect::<Vec<_>>().join(", ")
```

Comment:
How about this instead (and the two other occurences below)?
```suggestion
                indices.join(", ")
```

Reply:
:exploding_head: 
I don't know how I ended up with this, probably started modifying the iterator and then managed to forget that I was operating on a slice in a first place.